solr/modules/monitor/src/java/org/apache/solr/monitor/search/ReverseSearchComponent.java

```diff
@@ -0,0 +1,254 @@
+    rb.setQuery(new MatchAllDocsQuery());
+    mutableFilters.add(preFilterQuery);
+    var searcher = req.getSearcher();
+    MonitorQueryCache solrMonitorCache =
+        (SharedMonitorCache) searcher.getCache(SOLR_MONITOR_CACHE_NAME);
+    SolrMonitorQueryDecoder queryDecoder = SolrMonitorQueryDecoder.fromCore(req.getCore());
+    mutableFilters.add(
+        new MonitorPostFilter(
+            new SolrMonitorQueryCollector.CollectorContext(
+                solrMonitorCache, queryDecoder, matcherSink, writeToDocList, matchType)));
+    rb.setFilters(mutableFilters);
+    rb.rsp.add(QUERY_MATCH_TYPE_KEY, matchType.name());
+    if (matchType != QueryMatchType.NONE) {
+      rb.rsp.add(MONITOR_OUTPUT_KEY, monitorResult);
+    }
+  }
+
+  @SuppressWarnings({"unchecked"})
+  private DocumentBatchVisitor documentBatch(SolrQueryRequest req) {
+    Object jsonParams = req.getJSON().get("params");
+    if (!(jsonParams instanceof Map)) {
+      throw new SolrException(SolrException.ErrorCode.BAD_REQUEST, "need params");
+    }
+    var paramMap = (Map<?, ?>) jsonParams;
+    var documents = paramMap.get(MONITOR_DOCUMENTS_KEY);
+    if (!(documents instanceof List)) {
+      throw new SolrException(SolrException.ErrorCode.BAD_REQUEST, "need documents list");
+    }
+    List<Document> luceneDocs = new ArrayList<>();
+    for (var document : (List<?>) documents) {
+      if (!(document instanceof Map)
+          || !((Map<?, ?>) document).keySet().stream().allMatch(key -> key instanceof String)) {
+        throw new SolrException(
+            SolrException.ErrorCode.BAD_REQUEST, "document needs to be a string-keyed map");
+      }
+      var solrInputDoc = JsonLoader.buildDoc((Map<String, Object>) document);
+      var luceneDoc = DocumentBuilder.toDocument(solrInputDoc, req.getSchema());
+      luceneDocs.add(luceneDoc);
+    }
+    return DocumentBatchVisitor.of(req.getSchema().getIndexAnalyzer(), luceneDocs);
+  }
+
+  @Override
+  public void handleResponses(ResponseBuilder rb, ShardRequest sreq) {
+    if (skipReverseSearch(rb) || (sreq.purpose & ShardRequest.PURPOSE_GET_TOP_IDS) == 0) {
+      return;
+    }
+    var responses =
+        sreq.responses.stream()
+            .map(shardResponse -> shardResponse.getSolrResponse().getResponse())
+            .collect(Collectors.toList());
+    rb.rsp.getValues().removeAll(MONITOR_OUTPUT_KEY);
+    var matchType = QueryMatchType.fromString(rb.req.getParams().get(QUERY_MATCH_TYPE_KEY));
+    if (matchType != QueryMatchType.NONE) {
+      var finalOutput = mergeResponses(responses, matchType);
+      rb.rsp.add(MONITOR_OUTPUT_KEY, finalOutput);
+    }
+  }
+
+  @Override
+  public String getDescription() {
+    return "Component that integrates with lucene monitor for reverse search.";
+  }
+
+  private static boolean skipReverseSearch(ResponseBuilder rb) {
+    return !rb.req.getParams().getBool(REVERSE_SEARCH_PARAM_NAME, false);
+  }
```

Comment:
Wondering if this flag is still needed now that `ReverseSearchComponent` is extending `QueryComponent` -- https://github.com/cpoerschke/solr/commit/b51176b26b863da08f8deff4e382d95beea1acea and https://github.com/cpoerschke/solr/commit/844790fd673d307595a606977d40ad64a7e2105d on https://github.com/cpoerschke/solr/commits/solr-monitor-cpoerschke-1/ naively try to remove it but that's breaking the tests. Have not yet looked into further.

Reply:
This is because I overloaded `select`/`query` paths to run reverse search logic. The delete test relied on a regular "forward" search for its validation. I've changed the tests to most closely mirror how this module probably should be used, which is on a dedicated path. I am calling it `/reverseSearch` for now.